server/controllers/appointment.js

```diff
@@ -0,0 +1,48 @@
+const asyncHandler =  require('express-async-handler');
+const validatorResults = require('express-validator');
+const Appointment = require('../models/Appointment');
+const Meeting = require('../models/Meeting');
+const moment = require('moment-timezone');
+//CREATE /appointment -> Create an appointment
+//meeting_id, name, email, time (datetime), timezone
+//GET /appointments -> list of appointments for logged in user
+
+// @route  POST api/appointment
+// @des    Creates appointment
+// @access Public
+exports.createAppointment = asyncHandler(async (req, res) => {
+    const { name, email, time } = req.body;
+    const meeting = await Meeting.findOne({ meetingId: req.params.user_id });
+    
+    const appointmentFields = {};
+    appointmentFields.meetingId = meeting;
+    if(name) appointmentFields.name = name;
+    if(email) appointmentFields.email = email;
+    if(time) appointmentFields.time = time;
+    appointmentFields.timezone = moment.tz.guess();
+
+    const newAppointment = new Appointment(appointmentFields);
+    await newAppointment.save();
+    res.json(newAppointment);
```

Comment:
This works but it's not a clean approach. I would suggest we do some request body validation above. Like `if (!name && !email && !time) { . . . }` as name, email, and time are all required fields. Then instead of having appointmentFields we could do `new Appointment({ . . . })` and just pass the object directly.

Reply:
I cleaned this up as suggested and used the request body to fill out the properties.